x-pack/packages/kbn-elastic-assistant/impl/assistant/api.test.tsx

```diff
@@ -28,136 +35,277 @@ const messages: Message[] = [
+        {
+          method: 'GET',
+          signal: undefined,
+        }
+      );
     });
+    it('returns error when error is an error', async () => {
+      const testProps = {
+        resource: 'a-resource',
+        http: mockHttp,
+      };
+      const error = 'simulated error';
+      (mockHttp.fetch as jest.Mock).mockImplementation(() => {
+        throw new Error(error);
+      });
+
+      await expect(getKnowledgeBaseStatus(testProps)).resolves.toThrowError('simulated error');
+    });
+  });
 
-    const testProps: FetchConnectorExecuteAction = {
-      assistantLangChain: true, // <-- requires response parsing
-      http: mockHttp,
-      messages,
-      apiConfig,
-    };
+  describe('postKnowledgeBase', () => {
+    it('calls the knowledge base API when correct resource path', async () => {
+      const testProps = {
+        resource: 'a-resource',
+        http: mockHttp,
+      };
+
+      await postKnowledgeBase(testProps);
+
+      expect(mockHttp.fetch).toHaveBeenCalledWith(
+        '/internal/elastic_assistant/knowledge_base/a-resource',
+        {
+          method: 'POST',
+          signal: undefined,
+        }
+      );
+    });
+    it('returns error when error is an error', async () => {
+      const testProps = {
+        resource: 'a-resource',
+        http: mockHttp,
+      };
+      const error = 'simulated error';
+      (mockHttp.fetch as jest.Mock).mockImplementation(() => {
+        throw new Error(error);
+      });
+
+      await expect(postKnowledgeBase(testProps)).resolves.toThrowError('simulated error');
+    });
+  });
 
-    const result = await fetchConnectorExecuteAction(testProps);
+  describe('deleteKnowledgeBase', () => {
+    it('calls the knowledge base API when correct resource path', async () => {
+      const testProps = {
+        resource: 'a-resource',
+        http: mockHttp,
+      };
+
+      await deleteKnowledgeBase(testProps);
+
+      expect(mockHttp.fetch).toHaveBeenCalledWith(
+        '/internal/elastic_assistant/knowledge_base/a-resource',
+        {
+          method: 'DELETE',
+          signal: undefined,
+        }
+      );
+    });
+    it('returns error when error is an error', async () => {
+      const testProps = {
+        resource: 'a-resource',
+        http: mockHttp,
+      };
+      const error = 'simulated error';
+      (mockHttp.fetch as jest.Mock).mockImplementation(() => {
+        throw new Error(error);
+      });
+
+      await expect(deleteKnowledgeBase(testProps)).resolves.toThrowError('simulated error');
+    });
+  });
 
-    expect(result).toEqual({ response, isError: false });
+  describe('postEvaluation', () => {
+    it('calls the knowledge base API when correct resource path', async () => {
+      (mockHttp.fetch as jest.Mock).mockResolvedValue({ success: true });
+      const testProps = {
+        http: mockHttp,
+        evalParams: {
+          agents: ['not', 'alphabetical'],
```

Comment:
I don't think I actually had any reason for sorting these params before sending them over the wire 😅 Nice catch here! 😀